src/main/combatLogParser.ts

```diff
@@ -0,0 +1,434 @@
+            .split("\n");
+
+        return content.length > 1 ? content[1] : 'unknown';
+    }
+
+    /**
+     * Ensure only a single logfile is being watched once one of them starts
+     * receiving data.
+     *
+     * The lock will timeout after 30 seconds if no data has been received.
+     */
+    private lockHandler(handler: CombatLogMonitorHandlerType): boolean {
+        // If it's locked, and not by 'handler', get out.
+        if (this._handlerLock && this._handlerLock !== handler) {
+            return false;
+        }
+
+        // Reset timeout and return
+        if (this._handlerLock === handler) {
+            this.resetLockTimeout();
+            return true;
+        }
+
+        console.log(`[CombatLogParser] Locking path '${handler.path}' for exclusive event processing.`)
+
+        this.emit('DataFirstEvent', handler.wowFlavour);
+
+        this.resetLockTimeout();
+
+        this._handlerLock = handler;
+        return true;
+    }
+
+    /**
+     * Reset the lock/data timeout timer such that it doesn't fire when it
+     * isn't supposed to.
+     */
+    private resetLockTimeout(): void {
+        if (this._handlerLockTimeout) {
+            clearTimeout(this._handlerLockTimeout);
+        }
+
+        this._handlerLockTimeout = setTimeout(() => {
+            console.log(`[CombatLogParser] Unlocking path '${this._handlerLock?.path}' for exclusive event processing since no data received in ${this._options.dataTimeout / 1000} seconds.`)
+            this._handlerLock = undefined;
+            this.emit('DataTimeout', this._options.dataTimeout);
+        }, this._options.dataTimeout);
+    }
+
+    /**
+     * Find and return the most recent file that matches the combat log filename
+     * pattern.
+     */
+    private getLatestLog (pathSpec: string): string | undefined {
+        const globPath = path.join(pathSpec, 'WoWCombatLog*.txt');
+        const logs = glob.sync(globPath)
+            .map((name: string) => ({name, mtime: fs.statSync(name).mtime}))
+            .sort((A: any, B: any) => B.mtime.getTime() - A.mtime.getTime());
+
+        if (logs.length === 0) {
+            return;
+        }
+
+        return logs[0].name;
+    }
+
+    /**
+     * Monitor a file for new content and process combat log lines accordingly
+     * when they arrive.
+     */
+    private tailLogFile(handler: CombatLogMonitorHandlerType): void {
+        // Clear any old tail handler before creating the new instance
+        if (handler.tail) {
+            handler.tail.unwatch();
+        }
+
+        const tailHandler = new tail(handler.currentLogFile, this._tailOptions);
+        tailHandler
+            .on('line', (line: string) => {
+                if (!this.lockHandler(handler)) {
+                    return;
+                }
+
+                this.handleLogLine(handler.wowFlavour, line);
+            })
+            .on('error', (error: unknown) => {
+                console.error('[CombatLogParser] ERROR: ', error);
```

Comment:
Don't need the "ERROR" in here as it's already an error level log - but worth some text giving the context of "tailing log file"